### PR TITLE
update dotnet monitor to 7

### PR DESCRIPTION
**Pull request by waliMSFT**
update dotnet monitor to 7 for dotnet 6 and dotnet 7

### CODE REVIEW

**Review by snehapar9 — DISMISSED** [review]
LGTM

File: images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile
Comment (daniv-msft) [inline images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile:12]: @pauld-msft Do we even need this *staging* file anymore for .NET 7? Should this one just be removed?

Comment (pauld-msft) [reply images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile:12]: Yep it can be removed, I didn't initially delete it so it could be a sample, but we can always find it in the git history if we need to use it as a sample I suppose!

File: images/build/installDotNetCore.sh
Comment (waliMSFT) [reply images/build/installDotNetCore.sh:77]: to my understanding, this is the default dotnet-version we use for dockerfiles. Then an individual dotnet version takes the dotnet-monitor version from its dockerfile.
So far, when we update the dotnet-version to a new one we set the new version as default. 
Please let me know if you think otherwise. Thanks

File: images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile
Comment (waliMSFT) [reply images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile:12]: @pauld-msft should I remove it in my branch?

Comment (pauld-msft) [reply images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile:12]: @waliMSFT Sounds like we want it removed eventually and this is as good a time as any to do so! Thanks!

File: images/build/installDotNetCore.sh
Comment (waliMSFT) [reply images/build/installDotNetCore.sh:77]: @cormacpayne can you please confirm it?

Comment (cormacpayne) [reply images/build/installDotNetCore.sh:77]: This `installDotNetCore.sh` script is used by a few build image Dockerfiles to pre-bake a version of .NET Core into the image, and some will set the `INSTALL_PACKAGES` environment variable that causes the above section to execute. Looking at previous PRs where we modify the version of `dotnet-monitor` installed, it appears that we bump the version to the same value across all runtime images using it and this script, so it doesn't appear this would cause any issues from a version-"mismtatch" point of view, but would still be worth testing from our side and letting the Antares team know of the change so they can test as well.

The one other thing I'd check is that all other runtime images that have a step to install `dotnet-monitor` are using the same version since it appears we're just updating the 6.0 and 7.0 images, but the 3.1 and 5.0 images also have a step to install` dotnet-monitor` that will be stuck on some version of `6.*` or `6.1.*` moving forward.

File: images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile
Comment (waliMSFT) [reply images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile:12]: @pauld-msft I have removed the file as suggested. Just to know why this file was added previously?

Comment (pauld-msft) [reply images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile]: @waliMSFT this was added to support staging for dotnet 7 runtime images https://github.com/microsoft/Oryx/pull/1643#discussion_r1009498544

**Review by pauld-msft — CHANGES_REQUESTED** [review]
We will need to just update the runtime image base tags for these to actually roll out the changes to our runtime images, and which will need to be coordinated with https://github.com/microsoft/Oryx/pull/1677 - the steps to make sure both changes make it to the final base image are:
1. merge in https://github.com/microsoft/Oryx/pull/1677 to `main`
2. merge `main` into this branch
3. re-run the base images pipelines: https://devdiv.visualstudio.com/DevDiv/_build?definitionId=11092&_a=summary
4. Update the runtime base image tag in constants.yaml and run generateConstants

Other than that, these changes look good to me so I'll approve once we run through these steps

File: images/build/installDotNetCore.sh
Comment (cormacpayne) [reply images/build/installDotNetCore.sh:77]: Updated my comment above to differentiate `INSTALL_PACKAGES` and `INSTALL_TOOLS` -- it doesn't appear that `INSTALL_TOOLS` is being set anywhere in our repository, and it's not something that users can set as this is only relevant when the `installDotNetCore.sh` script is ran when constructing the images, so the above section is never currently executed.

Historically, when this was [first added](https://github.com/microsoft/Oryx/commit/a8047325620bddbf82b210a52d8caeecbd939239#diff-d2d91164bba96021bc9e98f865d8f0073c633c60aca54d7eafafcedd8ac29102), we were setting `INSTALL_TOOLS` in the `latest` Dockerfile when installing .NET Core 1.1 and 3.0. I think around [this commit](https://github.com/microsoft/Oryx/commit/58e488370675955a21da4c775dfa3efc356873d2#) was when the `INSTALL_TOOLS` environment variable stopped being set in the Dockerfile, but I can't find any reason for why.

If we haven't used it in ~3 years, it may not be a bad idea to remove this snippet from the `installDotNetCore.sh` script just so we're not worried about bumping versions here moving forward.

Comment (waliMSFT) [reply images/build/installDotNetCore.sh:77]: @cormacpayne  Thanks for the feedback. 
I will remove the whole code-block started from the if condition- `if [ "$INSTALL_TOOLS" == "true" ]; then`
https://github.com/microsoft/Oryx/blob/main/images/build/installDotNetCore.sh#:~:text=if%20%5B%20%22%24INSTALL_TOOLS,fi

Comment (waliMSFT) [reply images/build/installDotNetCore.sh]: @pauld-msft yah dotnet-monitor is tool is used only in the runtime images.

**Review by pauld-msft — DISMISSED** [review]
:shipit:

**Comment by cormacpayne** [comment]
@waliMSFT are we waiting on anything to merge this PR?

**Comment by waliMSFT** [comment]
> 

I have been waiting for our partners to confirm us about this. I followed up with them but have not received any feedback yet from them.

File: images/build/installDotNetCore.sh
Comment (pauld-msft) [reply images/build/installDotNetCore.sh:66]: INSTALL_TOOLS context: https://github.com/microsoft/Oryx/pull/1672#discussion_r1020421508

**Comment by waliMSFT** [comment]
Thanks @pauld-msft. I will make the needed changes.

**Review by pauld-msft — APPROVED** [review]
:shipit:

**Pull request merged**
